src/theme/progress.ts

```diff
@@ -0,0 +1,218 @@
+      stepsContainer: 'text-base'
+    }
+  }, {
+    orientation: 'vertical',
+    size: '2xs',
+    class: {
+      root: 'w-px',
+      statusContainer: 'flex-col min-h-fit',
+      status: 'text-xs',
+      stepsContainer: 'text-xs'
+    }
+  }, {
+    orientation: 'vertical',
+    size: 'xs',
+    class: {
+      root: 'w-0.5',
+      statusContainer: 'flex-col min-h-fit',
+      status: 'text-xs',
+      stepsContainer: 'text-xs'
+    }
+  }, {
+    orientation: 'vertical',
+    size: 'sm',
+    class: {
+      root: 'w-1',
+      statusContainer: 'flex-col min-h-fit',
+      status: 'text-sm',
+      stepsContainer: 'text-sm'
+    }
+  }, {
+    orientation: 'vertical',
+    size: 'md',
+    class: {
+      root: 'w-2',
+      statusContainer: 'flex-col min-h-fit',
+      status: 'text-sm',
+      stepsContainer: 'text-sm'
+    }
+  }, {
+    orientation: 'vertical',
+    size: 'lg',
+    class: {
+      root: 'w-3',
+      statusContainer: 'flex-col min-h-fit',
+      status: 'text-sm',
+      stepsContainer: 'text-sm'
+    }
+  }, {
+    orientation: 'vertical',
+    size: 'xl',
+    class: {
+      root: 'w-4',
+      statusContainer: 'flex-col min-h-fit',
+      status: 'text-base',
+      stepsContainer: 'text-base'
+    }
+  }, {
+    orientation: 'vertical',
+    size: '2xl',
+    class: {
+      root: 'w-5',
+      statusContainer: 'flex-col min-h-fit',
+      status: 'text-base',
+      stepsContainer: 'text-base'
+    }
+  }, {
+    orientation: 'horizontal',
+    animation: 'carousel',
+    class: { indicator: 'data-[state=indeterminate]:animate-[carousel_2s_ease-in-out_infinite]' }
+  }, {
+    orientation: 'vertical',
+    animation: 'carousel',
+    class: { indicator: 'data-[state=indeterminate]:animate-[carousel-vertical_2s_ease-in-out_infinite]' }
+  }, {
+    orientation: 'horizontal',
+    animation: 'carousel-inverse',
+    class: { indicator: 'data-[state=indeterminate]:animate-[carousel-inverse_2s_ease-in-out_infinite]' }
+  }, {
+    orientation: 'vertical',
+    animation: 'carousel-inverse',
+    class: { indicator: 'data-[state=indeterminate]:animate-[carousel-inverse-vertical_2s_ease-in-out_infinite]' }
+  }, {
+    orientation: 'horizontal',
+    animation: 'swing',
+    class: { indicator: 'data-[state=indeterminate]:animate-[swing_2s_ease-in-out_infinite]' }
+  }, {
+    orientation: 'vertical',
+    animation: 'swing',
+    class: { indicator: 'data-[state=indeterminate]:animate-[swing-vertical_2s_ease-in-out_infinite]' }
+  }, {
+    orientation: 'horizontal',
+    animation: 'elastic',
+    class: { indicator: 'data-[state=indeterminate]:animate-[elastic_2s_ease-in-out_infinite]' }
+  }, {
+    orientation: 'vertical',
+    animation: 'elastic',
+    class: { indicator: 'data-[state=indeterminate]:animate-[elastic-vertical_2s_ease-in-out_infinite]' }
+  }],
+  defaultVariants: {
+    animation: 'carousel',
+    color: 'primary',
+    orientation: 'horizontal',
+    size: 'md',
+    step: 'other'
```

Comment:
Does this need to be in `defaultVariants`? 🤔

Reply:
If you refer to `step`, since I already handle it in the `stepVariant` function indeed it isn't needed as a `deafultVariants`